### DISCUSSION TITLE
`position_encoding param` warning when moving in a file

### DISCUSSION THREAD

**Discussion by tembbo** [neovim/neovim#31495] [Q&A] [answered]
When I start to move around in a file in a rust file for the first time (since opening), i get this warning: `position_encoding param is required in vim.lsp.util.make_position_params. Defaulting to position encoding of the first cl
ient.` Thanks

**Comment by clason** [comment]
Please open an issue with all the required(!) information. (You're probably following `0.11` nightlies blindly without updating your config or your plugins to the breaking changes.)

**Reply by tembbo** [reply]
Thanks

**Reply by fredrikaverpil** [reply]
@realguse did you end up reporting an issue on this - or solve it?
If not, I can write something up as I am also experiencing this.

**Reply by tembbo** [reply]
@fredrikaverpil No I did in fact not make an issue out of this because I'm not too experienced with nvim so I didnt find a way to reproduce with a minimal config, but if you could write the issue that would be amazingbe, just make sure to link it to this

**Reply by fredrikaverpil** [reply]
@realguse I'm having a hard time reproducing this. Here's the `minimal.lua` I have so far, where I've added blink.cmp, fzf-lua and lua_ls.

```lua
-- Run this file as `nvim --clean -u minimal.lua`

-- NOTE: make sure lua-language-server is on $PATH
-- vim.env.PATH = vim.env.PATH .. ":" .. vim.fn.expand("~/.local/share/fredrik/mason/bin")

for name, url in pairs({
  -- ADD PLUGINS _NECESSARY_ TO REPRODUCE THE ISSUE, e.g:
  -- 'https://github.com/author1/plugin1',
  -- 'https://github.com/author2/plugin2',
  "https://github.com/saghen/blink.cmp",
  "https://github.com/ibhagwan/fzf-lua",
}) do
  local install_path = vim.fn.fnamemodify("nvim_issue/" .. name, ":p")
  if vim.fn.isdirectory(install_path) == 0 then
    vim.fn.system({ "git", "clone", "--depth=1", url, install_path })
  end
  vim.opt.runtimepath:append(install_path)
end

-- ADD INIT.LUA SETTINGS _NECESSARY_ FOR REPRODUCING THE ISSUE

-- blink.cmp
local opts = {
  completion = {
    list = { selection = "manual" },
    documentation = {
      auto_show = true,
    },
  },
  signature = {
    enabled = true, -- experimental, can also be provided by noice
  },

  -- default list of enabled providers defined so that you can extend it
  -- elsewhere in your config, without redefining it, via `opts_extend`
  sources = {
    default = { "lsp", "path", "snippets", "buffer" },
  },
}
require("blink.cmp").setup(opts)

-- fzf
require("fzf-lua").setup()

vim.keymap.set("n", "gd", "<cmd>FzfLua lsp_definitions jump_to_single_result=true ignore_current_line=true<cr>", { noremap = true, silent = true })

local lsp_name = "lua_ls"

---@type vim.lsp.Config
local config = {
  cmd = { "lua-language-server" },
  root_markers = { ".luarc.json", ".luarc.jsonc", ".luacheckrc", ".stylua.toml", "stylua.toml", "selene.toml", "selene.yml", ".git" },
  filetypes = { "lua" },
  settings = {
    Lua = {
      runtime = {
        version = "LuaJIT",
      },
      workspace = {
        checkThirdParty = false,
      },
      codeLens = {
        enable = false,
      },
      completion = {
        callSnippet = "Replace",
      },
      doc = {
        privateName = { "^_" },
      },
      hint = {
        enable = true,
        setType = false,
        paramType = true,
        paramName = "Disable",
        semicolon = "Disable",
        arrayIndex = "Disable",
      },
    },
  },
}

local client_capabilities = vim.tbl_deep_extend("force", vim.lsp.protocol.make_client_capabilities(), require("blink.cmp").get_lsp_capabilities())
config.capabilities = client_capabilities

vim.lsp.config[lsp_name] = config
vim.lsp.enable(lsp_name)
```

Open Neovim like this to invoke it with this minimal config:

```bash
nvim --clean -u minimal.lua
```

Then open up a lua file with `:e foo.lua`.

Make sure you have `lua-language-server` on $PATH.

<img width="935" alt="image" src="https://github.com/user-attachments/assets/753a879e-b01c-4a52-b4a0-5d578e007cc2" />

**Reply by tembbo** [reply]
This looks very good, it's unfortunate that submitting an issue in this repo is so strict, but I guess it's necessary to actually have a reasonable attempt at fixing them. I think you should go ahead and submit the issue since you know more about this than I do

**Reply by fredrikaverpil** [reply]
I found the culprit by trial and error: it's `RRethy/vim-illuminate`, so we shouldn't file a bug report in this repo.

Looks like someone is already on to merging a fix: https://github.com/RRethy/vim-illuminate/pull/218

**Reply by fredrikaverpil** [reply, marked as answer]
Looks like the PR has been opened for quite some time. Maybe it's time to look for an alternative solution. Here are some alternatives;

- https://github.com/echasnovski/mini.nvim/blob/main/readmes/mini-cursorword.md
- https://github.com/tzachar/local-highlight.nvim

or, use the not yet merged change to vim-illuminate:

```lua
return {
  {
    -- "RRethy/vim-illuminate", -- TODO: https://github.com/RRethy/vim-illuminate/pull/218
    "rockyzhang24/vim-illuminate",
    branch = "fix-encoding",
    lazy = true,
    event = "BufReadPost",
    config = function(_, opts)
      require("illuminate").configure(opts)
    end,
  },
}
```

**Comment by eboody** [comment]
```lua
-- Silence the specific position encoding message
local notify_original = vim.notify
vim.notify = function(msg, ...)
  if
    msg
    and (
      msg:match 'position_encoding param is required'
      or msg:match 'Defaulting to position encoding of the first client'
      or msg:match 'multiple different client offset_encodings'
    )
  then
    return
  end
  return notify_original(msg, ...)
end
```

I added this to my init.lua to silence those warnings..

**Reply by justinmk** [reply]
Then you are using plugins that have bugs.

**Reply by eboody** [reply]
Yeah, sadly.

The problem is that trouble.nvim had its fix merged but hasnt had a new release in almost a year. So until they do that this is a hack that silences the annoying warning. 

Sometimes neovim releases an update and does something that requires plugins make a change. Like deprecating something.

**Reply by devansh08** [reply]
trouble.nvim has this fixed in v3.7.1.

But in my case, seems like some other plugin is the problem. Any ideas on how to pinpoint the plugin for this ? Barring enabling only plugin at a time and checking...

Edit: It was telescope.nvim and because I was using my outdated fork :) If anyone is stuck on how to make it easier to find, I just grepped for `make_position_params` in `~/.local/share/nvim/lazy`.
